sessiontxn/isolation/readcommitted.go

```diff
@@ -251,7 +310,25 @@ func (p *PessimisticRCTxnContextProvider) AdviseOptimizeWithPlan(val interface{}
 		plan = execute.Plan
 	}
 
-	if v, ok := plan.(*plannercore.Insert); ok && v.SelectPlan == nil {
+	useLastOracleTS := false
+	sessionVars := p.sctx.GetSessionVars()
+	if !sessionVars.RetryInfo.Retrying {
+		if v, ok := plan.(*plannercore.Insert); ok && v.SelectPlan == nil {
+			if len(v.OnDuplicate) == 0 && !v.IsReplace {
+				useLastOracleTS = true
+			}
+		} else {
+			if sessionVars.ConnectionID > 0 && sessionVars.InTxn() {
+				useLastOracleTS = planSkipGetTsoFromPD(p.sctx, plan, false)
+			}
+		}
```

Comment:
Could we merge the insert part into `planSkipGetTsoFromPD`?

Reply:
I add the limitation " sessionVars.ConnectionID > 0 && sessionVars.InTxn()" for point-write statments. If remove these limitation, I think it should merge "inser" process to planSkipGetTsoFromPD